splunk_connect_for_snmp_poller/manager/poller.py

```diff
@@ -69,13 +71,39 @@ def should_process_current_line(self, host, version, community, profile, frequen
             host, version, community, profile, frequency
         )
 
-    def check_inventory(self):
+    def check_server_config(self):
+        server_config_file = self._args.config
+
+        is_config_changed = False
+        # detect the config changes
+        if (
+            os.stat(server_config_file, follow_symlinks=True).st_mtime
+            > self._config_mod_time
+        ):
+            logger.info("[-] Change in scheduler config detected, reloading")
+            self._config_mod_time = os.stat(
+                server_config_file, follow_symlinks=True
+            ).st_mtime
+            is_config_changed = True
+
+        # get server_config on the runtime
+        runtime_server_config = parse_config_file(server_config_file)
```

Comment:
Also you're reloading server_config_time every time.

Reply:
Removed it.:)